unittest/libmariadb/basic-t.c

```diff
@@ -310,7 +310,7 @@ static int use_utf8(MYSQL *my)
 
   while ((row= mysql_fetch_row(res)) != NULL)
   {
-    FAIL_IF(strcmp(row[0], "utf8"), "wrong character set");
+    FAIL_IF(strcmp(row[0], "utf8mb3"), "wrong character set");
```

Comment:
Will server versions < 10.6 return utf8mb3 or utf8? If latter one, this change will break travis tests (testing 10.2 - 10.5). Maybe better to check both: `FAIL_IF(strcmp(row, "utf8") && strcmp(row[0], "utf8mb3"), ...)`

Reply:
after reading MDEV-8334 more careful it seems that 10.7 or later will use utf8mb4 instead of utf8mb3.. In this case I think it is better, not to change existing tests (setting utf8mb3 instead of utf8) but to check the results depending on the server version:

```
const char *get_utf8_name(MYSQL *mysql)
{
  unsigned long server_version= mysql_get_server_version(mysql);

  if (server_version >= 100700)
    return "utf8mb4";
  if (server_version >= 100600)
    return "utf8mb3";
  return "utf8";
}
```
and for checking the character set

`FAIL_IF(strcmp(row[0], get_utf8_name(mysql), "Error message");`

Reply:
I hear we use 3.2 and above for server version 10.6 and up(and 3.1 for < 10.6). and this patch is to be added on 3.2. I am sorry, I don't quite understand why it would affect version <  10.6 if we are not making changes on 3.1